src/test/java/games/strategy/engine/lobby/server/login/LobbyLoginValidatorTest.java

```diff
@@ -291,95 +246,182 @@ public void shouldNotUpdatePasswordsWhenUserHasOnlyMd5CryptedPassword() {
 
-        thenAuthenticationShouldSucceed();
-        thenUserShouldNotBeCreated();
-        thenUserShouldNotBeUpdated();
-      }
-
-      @Test
-      public void shouldNotUpdatePasswordsWhenUserHasOnlyBcryptedPassword() {
-        givenUserHasBcryptedPassword();
-        givenAuthenticationWillUseMd5CryptedPasswordAndFail();
-
-        whenAuthenticating(givenAuthenticationResponse());
-
-        thenAuthenticationShouldFailWithMessage(LobbyLoginValidator.ErrorMessages.AUTHENTICATION_FAILED);
+        thenAuthenticationShouldFailWithMessage(LobbyLoginValidator.ErrorMessages.ANONYMOUS_AUTHENTICATION_FAILED);
         thenUserShouldNotBeCreated();
         thenUserShouldNotBeUpdated();
       }
 
       private ResponseGenerator givenAuthenticationResponse() {
         return challenge -> ImmutableMap.of(
-            LobbyLoginValidator.HASHED_PASSWORD_KEY, md5Crypt(PASSWORD),
+            LobbyLoginValidator.ANONYMOUS_LOGIN, Boolean.TRUE.toString(),
             LobbyLoginValidator.LOBBY_VERSION, LobbyServer.LOBBY_VERSION.toString());
       }
     }
 
-    @ExtendWith(MockitoExtension.class)
     @Nested
-    public final class WhenUsingCurrentClientTest extends AbstractTestCase {
-      @Test
-      public void shouldNotUpdatePasswordsWhenUserHasBothPasswords() {
-        givenUserExists();
-        givenUserHasMd5CryptedPassword();
-        givenUserHasBcryptedPassword();
-        givenAuthenticationWillUseObfuscatedPasswordAndSucceed();
-
-        whenAuthenticating(givenAuthenticationResponse());
-
-        thenAuthenticationShouldSucceed();
-        thenUserShouldNotBeCreated();
-        thenUserShouldNotBeUpdated();
+    public final class WhenUserDoesNotExistTest {
+      @ExtendWith(MockitoExtension.class)
+      @Nested
+      public final class WhenUsingLegacyClientTest extends AbstractTestCase {
+        @Test
+        public void shouldCreateNewUserWithOnlyMd5CryptedPassword() {
+          givenUserDoesNotExist();
+
+          whenAuthenticating(givenAuthenticationResponse());
+
+          thenAuthenticationShouldSucceed();
+          thenUserShouldBeCreatedWithMd5CryptedPassword();
+          thenUserShouldNotBeUpdatedWithBcryptedPassword();
+        }
+
+        private ResponseGenerator givenAuthenticationResponse() {
+          return challenge -> ImmutableMap.of(
+              LobbyLoginValidator.EMAIL_KEY, EMAIL,
+              LobbyLoginValidator.HASHED_PASSWORD_KEY, md5Crypt(PASSWORD),
+              LobbyLoginValidator.LOBBY_VERSION, LobbyServer.LOBBY_VERSION.toString(),
+              LobbyLoginValidator.REGISTER_NEW_USER_KEY, Boolean.TRUE.toString());
+        }
       }
 
-      @Test
-      public void shouldUpdateBcryptedPasswordWhenUserHasOnlyMd5CryptedPassword() {
-        givenUserExists();
-        givenUserHasMd5CryptedPassword();
-        givenUserDoesNotHaveBcryptedPassword();
-        givenAuthenticationWillUseMd5CryptedPasswordAndSucceed();
+      @ExtendWith(MockitoExtension.class)
```

Comment:
I'm not 100% sure, but I think Extensions are being inherited to nested classes

Reply:
(So it would be sufficient to add this annotation once to the root class is what I'm trying to say)

Reply:
You're right on this @ssoloff .  I'm a bit curious if there is a JSR for extensions to be passed to nested classes. Another way to aks that is whether this is something IDEA should support, or if Eclipse is doing something non-standard

Reply:
d75e3ddec